arangod/Indexes/Index.cpp

```diff
@@ -415,6 +415,22 @@ bool Index::Compare(VPackSlice const& lhs, VPackSlice const& rhs) {
         return false;
       }
     }
+  } else if (type == IndexType::TRI_IDX_TYPE_IRESEARCH_LINK) {
+    value = lhs.get("view");
+    if (value.isString()) {
+      if (arangodb::basics::VelocyPackHelper::compare(
+              value, rhs.get("view"), false) != 0) {
+        auto ls = value.copyString();
```

Comment:
It's theoretically possible that `rhs.get("view")` is not a string. Then the following use of `rhs.get("view").copyString()` will throw an exception. I am not sure if this case can happen in reality, but it's unclear to me right now if this is the intended behavior of the compare function.

Reply:
@jsteemann As far as I can see any call to `Slice::get(const char*)` may throw an exception since it constructs `std::string` under the hood.

Reply:
@jsteemann To be honest I don't understand the reason of having `std::string` based API for VelocyPack. For arangodb it requires to have tons of static strings in order to avoid copies. Wouldn't be better to use `StringRef` instead?

Reply:
@gnusi Well, assuming we complete the move to C++17 soon, `std::string_view` would be a good candidate.

Reply:
@danielhlarkin yes, but both `VelocyPack` and `Arangod` already have `StringRef`s :-)